### DISCUSSION TITLE
SQLite in the browser - in-memory only?

### DISCUSSION THREAD

**Discussion by ChristianWeyer** [unoplatform/uno#6425] [Q&A] [answered]
The SQLite sample over there is quite cool :-)
https://github.com/unoplatform/Uno.Samples/tree/master/UI/SQLiteSample

This is using SQLite in-memory, right?

What would be a use-case for this, though?
Thanks!

@jeromelaban

**Comment by jeromelaban** [comment, marked as answer]
It's not using SQLite in-memory, but rather actual file system APIs (from what SQLite is concerned). 

The way emscripten abstracts the filesystem is a bit different though, where the FS entries are loaded in memory an persisted regularly to an Indexed DB. This is likely to change in the future, once proper filesystem access is provided to WebAssembly modules.

**Reply by ChristianWeyer** [reply]
Thanks Jerome :-)

**Reply by ChristianWeyer** [reply]
When currently using IndexedDb in the background, this means that the browser can delete storage, based on different circumstances.
Would be great to see real file access being used, soon.
